axelor-sale/src/main/java/com/axelor/apps/sale/rest/SaleOrderRestController.java

```diff
@@ -51,15 +51,23 @@ public Response createSaleOrder(SaleOrderPostRequest requestBody)
       throws AxelorException, JsonProcessingException {
     RequestValidator.validateBody(requestBody);
     new SecurityCheck().createAccess(SaleOrder.class).check();
+    Boolean inAti = getInAti(requestBody.getInAti());
 
     SaleOrder saleOrder =
         Beans.get(SaleOrderGeneratorService.class)
             .createSaleOrder(
                 requestBody.fetchClientPartner(),
                 requestBody.fetchCompany(),
                 requestBody.fetchContact(),
-                requestBody.fetchCurrency());
+                requestBody.fetchCurrency(),
+                inAti);
 
     return ResponseConstructor.buildCreateResponse(saleOrder, new SaleOrderResponse(saleOrder));
   }
+
+  private static Boolean getInAti(String inAti) {
+    if (inAti.equals("true")) return true;
+    if (inAti.equals("false")) return false;
+    return null;
+  }
```

Comment:
NPE occurs if we don't specify inAti in the request.

This method should not be in the controller, please move it in a service, SaleOrderRestService for example.
Also it does not need to be static.